tools/isobuild/package-api.js

```diff
@@ -336,6 +337,29 @@ _.extend(PackageAPI.prototype, {
     this._addFiles("sources", paths, arch, fileOptions);
   },
 
+  mainModule(path, arch) {
+    arch = toArchArray(arch);
+    forAllMatchingArchs(arch, a => {
+      const filesForArch = this.files[a];
+      const source = {
+        relPath: path,
+        fileOptions: {
+          mainModule: true
+        }
+      };
+
+      const oldMain = filesForArch.main;
+      if (oldMain) {
+        // It's not an error to call api.mainModule multiple times, but
```

Comment:
I guess I was imagining you might do
```js
api.mainModule("main.js");
```
and then later
```js
api.mainModule("server_main.js", "server");
```
to "override" the main module for server code, leaving `main.js` as the entry point for `web.browser` and `web.cordova`.

I'll admit that kind of pattern isn't very compelling with only two or three architectures in play, but it might be a reasonable thing to do in the future?